cmd/jujud/agent/machine.go

```diff
@@ -1787,12 +1812,17 @@ func getLogSinkConfig(cfg agent.Config) (apiserver.LogSinkConfig, error) {
 	return result, nil
 }
 
-func getAuditLogConfig(cfg controller.Config) apiserver.AuditLogConfig {
-	return apiserver.AuditLogConfig{
+func getAuditLogConfig(cfg controller.Config, logDir string) auditlog.Config {
+	result := auditlog.Config{
 		Enabled:        cfg.AuditingEnabled(),
 		CaptureAPIArgs: cfg.AuditLogCaptureArgs(),
 		MaxSizeMB:      cfg.AuditLogMaxSizeMB(),
 		MaxBackups:     cfg.AuditLogMaxBackups(),
 		ExcludeMethods: cfg.AuditLogExcludeMethods(),
 	}
+	if result.Enabled {
```

Comment:
Does this create a new log file each time we simply read the config?
That doesn't seem right - this should be done outside of here.

Reply:
Yeah, I think you're right - I'll pull it up to the caller instead.